src/main/utility/interval_map.rs

```diff
@@ -197,6 +198,7 @@ impl<V: Clone> IntervalMap<V> {
         // after the splice end (after having clipped the end of any existing interval contained in
         // the range, above).
         let splice_end = match self.ends.binary_search(&end) {
+            Ok(i) if self.ends[i] <= end => i + 1,
```

Comment:
Maybe put the `self.ends[i] <= end` in brackets? I was really confused for a minute wondering what a backward arrow means in a rust match expression :)

Reply:
Done